### PR TITLE
add completion command

### PR DESCRIPTION
**Pull request by lalyos**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?
Add "completion" command for enhanced shell ux

### CODE REVIEW

File: internal/cli/command/completion/cmd.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"os"
+
+	"github.com/banzaicloud/banzai-cli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+// NewCompletionCommand returns a cobra command for `completion` subcommands.
+func NewCompletionCommand(banzaiCli cli.Cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "completion",
```
Comment (sagikazarmark) [inline internal/cli/command/completion/cmd.go]: I'd love to have support for ZSH as well. Generally cobra based CLIs (eg. kubectl) have separate commands for shells, eg:

- `banzai completion bash`
- `banzai completion zsh`

File: internal/cli/command/completion/cmd.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"os"
+
+	"github.com/banzaicloud/banzai-cli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+// NewCompletionCommand returns a cobra command for `completion` subcommands.
+func NewCompletionCommand(banzaiCli cli.Cli) *cobra.Command {
+	cmd := &cobra.Command{
```
Comment (sagikazarmark) [inline internal/cli/command/completion/cmd.go]: I wonder if this command should be hidden. AFAIK most cobra based CLIs hide it. Anyone?

Comment (lalyos) [reply internal/cli/command/completion/cmd.go]: kubectl doesn't hide it

File: internal/cli/command/completion/cmd.go (hunk 1 above)
Comment (lalyos) [reply internal/cli/command/completion/cmd.go]: Please check the new version:

```
banzai completion 
Generates shell completion scripts

Usage:
  banzai completion [command]

Available Commands:
  bash        Generates bash completion scripts
  zshell      Generates zsh completion scripts
```

File: internal/cli/command/completion/cmd.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			Long: `To load completion run
+			
+			. <(banzai completion bash)
+			`,
+			Run: func(c *cobra.Command, args []string) {
+				c.GenBashCompletion(os.Stdout)
+			},
+		},
+		&cobra.Command{
+			Use:   "zshell",
```
Comment (sagikazarmark) [inline internal/cli/command/completion/cmd.go]:
```suggestion
			Use:   "zsh",
```

File: internal/cli/command/completion/cmd.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			Short: "Generates zsh completion scripts",
+			Long: `To load completion run
+				
+				. <(banzai completion zshell)
```
Comment (sagikazarmark) [inline internal/cli/command/completion/cmd.go]:
```suggestion
				. <(banzai completion zsh)
```

**Pull request merged**